src/Features/Blockcore.Features.Consensus/Rules/CommonRules/SetActivationDeploymentsPartialValidationRule.cs

```diff
@@ -1,18 +1,37 @@
 ﻿using System.Threading.Tasks;
 using Blockcore.Consensus.Rules;
+using Blockcore.Utilities.Store;
 
 namespace Blockcore.Features.Consensus.Rules.CommonRules
 {
     /// <summary>Set the <see cref="RuleContext.Flags"/> property that defines what deployments have been activated.</summary>
     public class SetActivationDeploymentsPartialValidationRule : PartialValidationConsensusRule
     {
+        private readonly IKeyValueRepository keyValueRepository;
+
+        public SetActivationDeploymentsPartialValidationRule()
+        {
+
+        }
+
+        public SetActivationDeploymentsPartialValidationRule(IKeyValueRepository keyValueRepository)
+        {
+            this.keyValueRepository = keyValueRepository;
+        }
+
         /// <inheritdoc />
         /// <exception cref="ConsensusErrors.InvalidPrevTip">The tip is invalid because a reorg has been detected.</exception>
         public override Task RunAsync(RuleContext context)
         {
             // Calculate the consensus flags and check they are valid.
             context.Flags = this.Parent.NodeDeployments.GetFlags(context.ValidationContext.ChainedHeaderToValidate);
 
+            if (this.keyValueRepository != null)
```

Comment:
hmm I am not sure this is a good idea, this means we push to disk on each block arriving.
It's not such a good idea to push to disk in the rules in general, perhaps this should be done on shutdown however then we need to check for crashes (check the node deployment current height is the same as the chained header.

I think we can persist to disk at the same time we persist the chained headers in the base feature (this happens every few min).